### DISCUSSION TITLE
how to set up webhooks

### DISCUSSION THREAD

**Discussion by D4rkT1m0** [firefly-iii/firefly-iii#9306] [Firefly III]
Hello guys,
I think i'm really interested in using webhooks.
but i really don't know where to start.
i got a synology nas with docker. my firefly instance is installed this way.

do you know what docker container i could install to get a "webhooks response server"?

a use case i would like to have is, when I create a transaction with a specific currency, the trigerred webhook will "manage" to convert transaction amount in my currency.
i suppose i have to get a webhook response server that will be able to play a script to calculate the new amout, to select the new curren y and to convert or create the transaction in FF3.

thank you for your help
Guillaume

**Comment by JC5** [comment]
> do you know what docker container i could install to get a "webhooks response server"?

There is no real Docker container for this, you would have to develop something. There are services like [Pipedream](https://pipedream.com/) who can do bits and pieces, but it will need some custom development. Are you a programmer, by any chance?

**Reply by D4rkT1m0** [reply]
Hello James, thank you for the answer. 
I'm not à developper, but I think I can manage to achieve some things with the help of AI and some friend of mine which are developers. 
I'll take a look at pipedream and make a feedback

**Comment by molnarti** [comment]
i have a related question:
what can i achieve with the response given to the webhook? is the response payload somehow processed by firefly, or it just uses  the response  code to indicate if the webhook succeeded or not? 

is it possible to trigger webhooks just for specific transactions?  i have a quite simple use case that triggers only for a specific transaction on a specific account - a single transaction in a month. but firefly triggers the webhook for each transaction created, which clutters my execution log in n8n

**Reply by JC5** [reply]
>  what can i achieve with the response given to the webhook? is the response payload somehow processed by firefly, or it just uses the response code to indicate if the webhook succeeded or not?

https://docs.firefly-iii.org/how-to/firefly-iii/features/webhooks/

> is it possible to trigger webhooks just for specific transactions? 

No, the idea is that you use webhooks when YOU want to do the distinction and processing.

**Reply by molnarti** [reply]
> https://docs.firefly-iii.org/how-to/firefly-iii/features/webhooks/
yep, i read that page, but its still not fully clear to me. 

1. Firefly sends a POST request to the defined URL target with a payload based on the webook settings (e.g. relevant details, account details, etc). This is greatly described in the https://docs.firefly-iii.org/how-to/firefly-iii/features/webhooks/#payload section
2. The target responds to Firefly with a payload in the POST response. **What can i put in the webhook POST response payload and how will Firefly react on it?** For example, based on the response i want to edit a transaction in Firefly, e.g. adding an external URL to it. This can be done by a separate API call, but I am wondering if i could do something similar directly in the webhook response, without a separate API call

**Reply by JC5** [reply]
You cannot. The webhook is "fire and forget". If you want to respond, you need an API call.
